contracts/token/ERC721/ERC721.sol

```diff
@@ -174,7 +174,24 @@ contract ERC721 is ERC165, IERC721 {
      * @param _data bytes data to send along with a safe transfer check
      */
     function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data) public {
-        transferFrom(from, to, tokenId);
+        require(_isApprovedOrOwner(msg.sender, tokenId), "ERC721: transfer caller is not owner nor approved");
```

Comment:
@nventuro Do we still have the guideline that every error message should be unique in the codebase?

If so this message should say something like `safe transfer`, so as to be diferent from the message in `_transferFrom`.